packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -0,0 +1,152 @@
+
+        replaceLwcImport(path, state);
+        catalogTmplImport(path, state);
+        catalogStyleImport(path, state);
+    },
+    ExportDefaultDeclaration(_path, _state) {},
+    ClassDeclaration(path, state) {
+        if (
+            path.node?.superClass?.type === 'Identifier' &&
+            // It is possible to inherit from something that inherits from
+            // LightningElement, so the detection here needs additional work.
+            path.node?.superClass?.name === 'LightningElement'
+        ) {
+            state.isLWC = true;
+            if (path.node!.id) {
+                state.lwcClassName = path.node!.id.name;
+            } else {
+                path.node!.id = b.identifier('DefaultComponentName');
+                state.lwcClassName = 'DefaultComponentName';
+            }
+        }
+    },
+    PropertyDefinition(path, state) {
+        if (
+            path.node &&
+            path.node.static &&
+            is.identifier(path.node.key) &&
+            path.node.key.name === 'stylesheets' &&
+            is.arrayExpression(path.node.value) &&
+            path.node.value.elements.every((el) => is.identifier(el))
+        ) {
+            catalogStaticStylesheets(
+                path.node.value.elements.map((el) => (el as EsIdentifier).name),
+                state
+            );
+        }
+    },
+    MethodDefinition(path, state) {
+        if (path.node?.key.type !== 'Identifier') {
+            return;
+        }
+        switch (path.node.key.name) {
+            case 'constructor':
+                path.node.value.params = [b.identifier('propsAvailableAtConstruction')];
+                break;
+            case 'connectedCallback':
+                state.hasConnectedCallback = true;
+                break;
+            case 'render':
+                state.hasRenderMethod = true;
+                break;
+            case 'renderedCallback':
+                state.hadRenderedCallback = true;
+                path.remove();
+                break;
+            case 'disconnectedCallback':
+                state.hadDisconnectedCallback = true;
+                path.remove();
+                break;
+            case 'errorCallback':
+                state.hadErrorCallback = true;
+                path.remove();
+                break;
+        }
+    },
+    Super(path, _state) {
+        const parentFn = path.getFunctionParent();
+        if (
+            parentFn &&
+            parentFn.parentPath?.node?.type === 'MethodDefinition' &&
+            parentFn.parentPath?.node?.kind === 'constructor' &&
+            path.parentPath &&
+            path.parentPath.node?.type === 'CallExpression'
+        ) {
+            path.parentPath.node.arguments = [b.identifier('propsAvailableAtConstruction')];
+        }
+    },
+};
+
+export default function compileJS(src: string, filename: string) {
+    const ast = parseModule(src, {
+        module: true,
+        next: true,
+    }) as EsProgram;
```

Comment:
It's a shame meriyah doesn't use `@types/estree` but exposes it's own types which are slightly different. These subtle differences could end up causing bigger issues.

Reply:
Based on [this issue](https://github.com/meriyah/meriyah/issues/199) it sounds like there's openness to refactoring on top of estree types; it's just a lot of work. The thing is, acorn also doesn't use estree types and instead reimplements them. So it seems like this is just a thing with JS parsers.

I may take a peak at the meriyah source to see how hard it'd be to contribute this change. And long term, despite the perf wins with meriyah, we can always use acorn like we're doing elsewhere.

Reply:
I'm also a bit concerned about having so many JS parsers in our packages in general. We already have Babel and Acorn and estree-walker, now we would be adding meriyah. Just an opportunity for any one of those things to throw breaking changes at us over time.

Reply:
I remember why I brought in meriyah - acorn doesn't support Js features until they are stage 4. This includes decorators. So it was a choice between babel+acorn or meriyah. I hate the double-transform, especially when compilation might be in the hot path, which is why I went with meriyah. I'm open to alternatives.